website.py

```diff
@@ -16,299 +16,303 @@ def print_headers(content_type: str) -> None:
-contextEl.addEventListener("change", function() {
-    localStorage.mips_to_c_saved_context = contextEl.value;
-});
-document.getElementById("options").addEventListener("change", function(event) {
-    var shouldSave = ["usesidebar", "allman", "leftptr", "globals", "nocasts", "noandor", "dark", "regvarsselect", "regvars"];
-    var options = {};
-    for (var key of shouldSave) {
-        var el = document.getElementsByName(key)[0];
-        options[key] = (el.type === "checkbox" ?
-            (el.checked ? "yes" : "no") :
-            el.value);
-    }
-    localStorage.mips_to_c_saved_options = JSON.stringify(options);
-});
+    sourceEl.addEventListener("change", function() {
+        localStorage.mips_to_c_saved_source = sourceEl.value;
+    });
+    contextEl.addEventListener("change", function() {
+        localStorage.mips_to_c_saved_context = contextEl.value;
+    });
+    document.getElementById("options").addEventListener("change", function(event) {
+        var shouldSave = ["usesidebar", "allman", "leftptr", "globals", "nocasts", "noandor", "dark", "regvarsselect", "regvars"];
+        var options = {};
+        for (var key of shouldSave) {
+            var el = document.getElementsByName(key)[0];
+            options[key] = (el.type === "checkbox" ?
+                (el.checked ? "yes" : "no") :
+                el.value);
+        }
+        localStorage.mips_to_c_saved_options = JSON.stringify(options);
+    });
 
-function updateSidebar() {
-    document.body.classList.toggle("has-sidebar", sidebarCheckboxEl.checked);
-    formEl.setAttribute("target", sidebarCheckboxEl.checked ? "outputframe" : "");
-    try {
-        document.getElementsByName("outputframe")[0].contentDocument.body.innerHTML = "";
-    } catch (e) {}
-}
-updateSidebar();
-sidebarCheckboxEl.addEventListener("change", updateSidebar);
-formEl.addEventListener("submit", function() {
-    if (!sidebarCheckboxEl.checked) return;
-    try {
-        document.getElementsByName("outputframe")[0].contentDocument.body.innerHTML = "Decompiling...";
-    } catch (e) {}
-});
-</script>
-</form>
-</body>
-</html>
-"""
-    )
+    function updateSidebar() {
+        document.body.classList.toggle("has-sidebar", sidebarCheckboxEl.checked);
+        formEl.setAttribute("target", sidebarCheckboxEl.checked ? "outputframe" : "");
+        try {
+            document.getElementsByName("outputframe")[0].contentDocument.body.innerHTML = "";
+        } catch (e) {}
+    }
+    updateSidebar();
+    sidebarCheckboxEl.addEventListener("change", updateSidebar);
+    formEl.addEventListener("submit", function() {
+        if (!sidebarCheckboxEl.checked) return;
+        try {
+            document.getElementsByName("outputframe")[0].contentDocument.body.innerHTML = "Decompiling...";
+        } catch (e) {}
+    });
+    </script>
+    </form>
+    </body>
+    </html>
+    """
+        )
+except:
+    # Set the headers for the cgitb traceback
+    print_headers(content_type="text/html")
```

Comment:
should only do this if headers haven't been sent; alternatively, try-except only the subprocess parts